ios/browser/api/net/certificate_utility.mm

```diff
@@ -142,15 +144,21 @@ + (int)verifyTrust:(SecTrustRef)trust
     }
 
     std::vector<base::ScopedCFTypeRef<SecCertificateRef>> intermediates;
-    for (CFIndex i = 1; i < cert_count; ++i) {
-      intermediates.emplace_back(SecTrustGetCertificateAtIndex(trust, i),
-                                 base::scoped_policy::RETAIN);
-    }
 
+    base::ScopedCFTypeRef<CFArrayRef> certificateChain(
+        SecTrustCopyCertificateChain(trust));
+    for (CFIndex i = 1; i < cert_count; i++) {
+      SecCertificateRef secCertificate =
+          base::mac::CFCastStrict<SecCertificateRef>(
+              CFArrayGetValueAtIndex(certificateChain, i));
+      intermediates.emplace_back(secCertificate, base::scoped_policy::RETAIN);
+    }
+    SecCertificateRef secCertificate =
+        base::mac::CFCastStrict<SecCertificateRef>(
+            CFArrayGetValueAtIndex(certificateChain, 0));
     return net::x509_util::CreateX509CertificateFromSecCertificate(
-        /*root_cert=*/base::ScopedCFTypeRef<SecCertificateRef>(
-            SecTrustGetCertificateAtIndex(trust, 0),
-            base::scoped_policy::RETAIN),
+        base::ScopedCFTypeRef<SecCertificateRef>(secCertificate,
+                                                 base::scoped_policy::RETAIN),
```

Comment:
Upstream changes seem to be implemented differently for iPhone < 15 and >= 15, do we need similar checks?